addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -661,23 +716,15 @@ def _import_fattura_pa(self, tree, invoice):
                             natura_element = element.xpath('.//Natura')
                             invoice_line_form.tax_ids.clear()
                             if percentage is not None:
-                                if natura_element and natura_element[0].text:
-                                    l10n_it_kind_exoneration = natura_element[0].text
-                                    tax = self.env['account.tax'].search([
-                                        ('company_id', '=', invoice_form.company_id.id),
-                                        ('amount_type', '=', 'percent'),
-                                        ('type_tax_use', '=', 'purchase'),
-                                        ('amount', '=', percentage),
-                                        ('l10n_it_kind_exoneration', '=', l10n_it_kind_exoneration),
-                                    ], limit=1)
-                                else:
-                                    tax = self.env['account.tax'].search([
-                                        ('company_id', '=', invoice_form.company_id.id),
-                                        ('amount_type', '=', 'percent'),
-                                        ('type_tax_use', '=', 'purchase'),
-                                        ('amount', '=', percentage),
-                                    ], limit=1)
-                                    l10n_it_kind_exoneration = ''
+                                l10n_it_kind_exoneration = natura_element and natura_element[0].text
+                                tax = self.env['account.tax'].search([
+                                    ('company_id', '=', invoice_form.company_id.id),
+                                    ('amount_type', '=', 'percent'),
+                                    ('amount', '=', percentage)]
+                                    + buyer_seller_info[incoming]['type_tax_use_domain']
+                                    + ([('l10n_it_kind_exoneration', '=', l10n_it_kind_exoneration)] if l10n_it_kind_exoneration else []),
```

Comment:
Hmm, in the changes of  @lordkrandel we always include the term. (so it is False), but this seems to follow the old logic.

Reply:
Right, this has to be changed.
`l10n_it_kind_exoneration` always has to be checked.
If it's not present in the XML, the taxes with it must be excluded.

Reply:
Do I understand it correctly that it should be kept like it is right now on this PR?